src/main/java/com/google/sps/servlets/SpecificGuideServlet.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Error;
+import com.google.sps.data.Guide;
+import com.google.sps.storage.DatastoreGuides;
+
+@WebServlet("/guides/*")
+public class SpecificGuideServlet extends HttpServlet{
+    static final long serialVersionUID = 0;
+    private final DatastoreGuides helper = new DatastoreGuides();
+    private final Gson gson = new Gson();
+    private long id;
+
+    private Optional<Error> getIdFromPath(String path){
+        String[] parts = path.split("/");
+        try{
+            id = Long.parseLong(parts[1]);
+        }catch(RuntimeException exception){
```

Comment:
You generally want to be (a little) more specific with exception types that you catch.

In this case, I think throwing an IllegalArgumentException from this method captures your intent pretty well. It's reasonable default behavior for any servlet to convert an IllegalArgumentException to an HTTP 400. So, you could re-throw the IndexOutOfBoundsException as an IllegalArgumentException (or instead of catching, check the array length and throw the error [1]). Then, let the doGet method catch IllegalArgumentException and convert it to a 400.

[1] At Google, we would do this with a precondition: `checkArgument(parts.length == 2);`

https://guava.dev/releases/21.0/api/docs/com/google/common/base/Preconditions.html#checkArgument-boolean-java.lang.Object-